blueprints/hr_assistant/date.py

```diff
@@ -0,0 +1,335 @@
+# -*- coding: utf-8 -*-
+"""This module contains the dialogue states for the 'date' domain in
+the MindMeld HR assistant blueprint application
+"""
+from .root import app
+from hr_assistant.general import (
+    _resolve_categorical_entities, _resolve_function_entity,
+    _resolve_extremes, _agg_function, _get_names, NOT_AN_EMPLOYEE,
+    SIZE
+)
+from dateutil.relativedelta import relativedelta
+import datetime
+import re
+
+
+@app.handle(intent='get_date')
+def get_date(request, responder):
+    """
+    If a user asks for a date related information of any person, this function returns
+    the required date. In case of a termination date related query, it also informs the user
+    of the reason for termination. For non-terminated employees, it informs the user about
+    the active current state of the employee in question.
+    """
+
+    # Look for name from the context of previous turn
+    name = request.frame.get('name')
+
+    # If user presents a new name, update the name in context
+    try:
+        name_ent = [e for e in request.entities if e['type'] == 'name']
+        name = name_ent[0]['value'][0]['cname']
+    except IndexError:
+        name = None
+
+    # If neither context nor the current query has an employee name, return not an employee
+    if not name:
+        responder.reply(NOT_AN_EMPLOYEE)
+        responder.listen()
+        return
+
+    # If name is found but not in the database, return not an employee
+    if name == '':
+        responder.reply(NOT_AN_EMPLOYEE)
+
+    responder.slots['name'] = name
+    responder.frame['name'] = name
+
+    employee = app.question_answerer.get(index='employee_data', emp_name=name)[0]
+
+    # 'action' entities represent employment action such as hiring or termination
+    action_entity = [e for e in request.entities if e['type'] == 'employment_action']
+
+    # 'dob' entities represent the date of birth entity
+    dob_entity = [e for e in request.entities if e['type'] == 'dob']
+
+    # Search for the determined action entity or date of birth in the KB and return.
+    # 'doh' - 'date of hire', 'dot' - 'date of termination', 'dob' - 'date of birth'
+    if action_entity:
+        action_entity = action_entity[0]['value'][0]['cname']
+
+        if action_entity == 'hired':
+            date = employee['doh']
+            responder.slots['date'] = date
+            responder.reply("{name}'s date of hire was {date}")
```

Comment:
I think you have to use string formats
```suggestion
            responder.reply(f"{name}'s date of hire was {date}")
```

Reply:
Ah, nvm.